### ISSUE TITLE
reactor stalls of 2.5 seconds  on all nodes after decommission starts

### ISSUE THREAD

**Issue by bentsi** [scylladb/scylla#5000]
*Installation details*
Scylla version (or git commit hash): 3.1.0.rc6-0.20190908.2ec036f50
Cluster size: 4
OS (RHEL/CentOS/Ubuntu/AWS AMI):  ami-0f02b94adb41f1f9c [eu-west-1]

dblog
```
Sep 10 09:51:30 ip-10-0-132-229 scylla: [shard 12] compaction - Compacted 11 sstables to [/var/lib/scylla/data/keyspace1/standard1-d23e5970d31b11e9a928000000000005/mc-169034-big-Data.db:level=1, /var/lib/scylla/data/keyspace1/standard1-d23e5970d31b11e9a928000000000005/mc-169048-big-Data.db:level=1, /var/lib/scylla/data/keyspace1/standard1-d23e5970d31b11e9a928000000000005/mc-169062-big-Data.db:level=1, /var/lib/scylla/data/keyspace1/standard1-d23e5970d31b11e9a928000000000005/mc-169076-big-Data.db:level=1, /var/lib/scylla/data/keyspace1/standard1-d23e5970d31b11e9a928000000000005/mc-169090-big-Data.db:level=1, /var/lib/scylla/data/keyspace1/standard1-d23e5970d31b11e9a928000000000005/mc-169104-big-Data.db:level=1, /var/lib/scylla/data/keyspace1/standard1-d23e5970d31b11e9a928000000000005/mc-169118-big-Data.db:level=1, /var/lib/scylla/data/keyspace1/standard1-d23e5970d31b11e9a928000000000005/mc-169132-big-Data.db:level=1, /var/lib/scylla/data/keyspace1/standard1-d23e5970d31b11e9a928000000000005/mc-169146-big-Data.db:level=1, /var/lib/scylla/data/keyspace1/standard1-d23e5970d31b11e9a928000000000005/mc-169160-big-Data.db:level=1, /var/lib/scylla/data/keyspace1/standard1-d23e5970d31b11e9a928000000000005/mc-169174-big-Data.db:level=1, /var/lib/scylla/data/keyspace1/standard1-d23e5970d31b11e9a928000000000005/mc-169188-big-Data.db:level=1, /var/lib/scylla/data/keyspace1/standard1-d23e5970d31b11e9a928000000000005/mc-169202-big-Data.db:level=1, ]. 2214209477 bytes to 2205761873 (~99% of original) in 47072ms = 44.69MB/s. ~2048384 total partitions merged to 2039025.
Sep 10 09:51:30 ip-10-0-132-229 scylla: [shard 12] compaction - Compacting [/var/lib/scylla/data/keyspace1/standard1-d23e5970d31b11e9a928000000000005/mc-169174-big-Data.db:level=1, ]
Sep 10 09:51:30 ip-10-0-132-229 scylla: Reactor stalled for 2470 ms on shard 1.
Sep 10 09:51:30 ip-10-0-132-229 scylla: Backtrace:
Sep 10 09:51:30 ip-10-0-132-229 scylla: 0x0000000004183e22
Sep 10 09:51:30 ip-10-0-132-229 scylla: 0x0000000004071135
Sep 10 09:51:30 ip-10-0-132-229 scylla: 0x0000000004071675
Sep 10 09:51:30 ip-10-0-132-229 scylla: 0x00000000040719b7
Sep 10 09:51:30 ip-10-0-132-229 scylla: 0x00007f3d524a202f
Sep 10 09:51:30 ip-10-0-132-229 scylla: 0x0000000000efbb33
Sep 10 09:51:30 ip-10-0-132-229 scylla: 0x0000000000d78545
Sep 10 09:51:31 ip-10-0-132-229 scylla: 0x0000000000d95e3b
Sep 10 09:51:31 ip-10-0-132-229 scylla: 0x0000000000d9676a
Sep 10 09:51:31 ip-10-0-132-229 scylla: 0x0000000000daebdf
Sep 10 09:51:31 ip-10-0-132-229 scylla: 0x0000000000db3501
Sep 10 09:51:31 ip-10-0-132-229 scylla: 0x000000000406e601
Sep 10 09:51:31 ip-10-0-132-229 scylla: 0x000000000406e81e
Sep 10 09:51:31 ip-10-0-132-229 scylla: 0x00000000041505ad
Sep 10 09:51:31 ip-10-0-132-229 scylla: 0x000000000417ac1b
Sep 10 09:51:31 ip-10-0-132-229 scylla: 0x000000000403c8dd
Sep 10 09:51:31 ip-10-0-132-229 scylla: /opt/scylladb/libreloc/libpthread.so.0+0x000000000000858d
Sep 10 09:51:31 ip-10-0-132-229 scylla: /opt/scylladb/libreloc/libc.so.6+0x00000000000fd6a2
Sep 10 09:51:31 ip-10-0-132-229 scylla: [shard 7] compaction - Compacted 13 sstables to [/var/lib/scylla/data/keyspace1/standard1-d23e5970d31b11e9a928000000000005/mc-172963-big-Data.db:level=3, /var/lib/scylla/data/keyspace1/standard1-d23e5970d31b11e9a928000000000005/mc-172977-big-Data.db:level=3, /var/lib/scylla/data/keyspace1/standard1-d23e5970d31b11e9a928000000000005/mc-172991-big-Data.db:level=3, /var/lib/scylla/data/keyspace1/standard1-d23e5970d31b11e9a928000000000005/mc-173005-big-Data.db:level=3, /var/lib/scylla/data/keyspace1/standard1-d23e5970d31b11e9a928000000000005/mc-173019-big-Data.db:level=3, /var/lib/scylla/data/keyspace1/standard1-d23e5970d31b11e9a928000000000005/mc-173033-big-Data.db:level=3, /var/lib/scylla/data/keyspace1/standard1-d23e5970d31b11e9a928000000000005/mc-173047-big-Data.db:level=3, /var/lib/scylla/data/keyspace1/standard1-d23e5970d31b11e9a928000000000005/mc-173061-big-Data.db:level=3, /var/lib/scylla/data/keyspace1/standard1-d23e5970d31b11e9a928000000000005/mc-173075-big-Data.db:level=3, /var/lib/scylla/data/keyspace1/standard1-d23e5970d31b11e9a928000000000005/mc-173089-big-Data.db:level=3, /var/lib/scylla/data/keyspace1/standard1-d23e5970d31b11e9a928000000000005/mc-173103-big-Data.db:level=3, ]. 1934722442 bytes to 1825926524 (~94% of original) in 37952ms = 45.88MB/s. ~1789696 total partitions merged to 1688588.

```
translated backtrace
```
void seastar::backtrace<seastar::backtrace_buffer::append_backtrace()::{lambda(seastar::frame)#1}>(seastar::backtrace_buffer::append_backtrace()::{lambda(seastar::frame)#1}&&) at /usr/include/boost/program_options/variables_map.hpp:146
seastar::backtrace_buffer::append_backtrace() at /usr/include/boost/program_options/variables_map.hpp:146
 (inlined by) print_with_backtrace at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/build/release/../../src/core/reactor.cc:1104
seastar::internal::cpu_stall_detector::generate_trace() at /usr/include/boost/program_options/variables_map.hpp:146
seastar::internal::cpu_stall_detector::maybe_report() at /usr/include/boost/program_options/variables_map.hpp:146
 (inlined by) seastar::internal::cpu_stall_detector::on_signal() at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/build/release/../../src/core/reactor.cc:1480
 (inlined by) seastar::reactor::block_notifier(int) at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/build/release/../../src/core/reactor.cc:1603
?? ??:0
mutation_partition::clear_gently(cache_tracker*) at crtstuff.c:?
 (inlined by) boost::intrusive::bstree_algorithms_base<boost::intrusive::rbtree_node_traits<void*, true> >::next_node(boost::intrusive::compact_rbtree_node<void*>* const&) at /usr/include/boost/intrusive/detail/bstree_algorithms_base.hpp:54
 (inlined by) boost::intrusive::tree_iterator<boost::intrusive::mhtraits<rows_entry, intrusive_set_external_comparator_member_hook, &rows_entry::_link>, true>::operator++() at /usr/include/boost/intrusive/detail/tree_iterator.hpp:86
 (inlined by) intrusive_set_external_comparator<rows_entry, &rows_entry::_link>::const_iterator::operator++() at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/intrusive_set_external_comparator.hh:91
 (inlined by) intrusive_set_external_comparator<rows_entry, &rows_entry::_link>::erase(intrusive_set_external_comparator<rows_entry, &rows_entry::_link>::const_iterator) at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/intrusive_set_external_comparator.hh:195
 (inlined by) intrusive_set_external_comparator<rows_entry, &rows_entry::_link>::iterator intrusive_set_external_comparator<rows_entry, &rows_entry::_link>::erase_and_dispose<auto current_deleter<rows_entry>()::{lambda(rows_entry*)#1}>(intrusive_set_external_comparator<rows_entry, &rows_entry::_link>::const_iterator, auto current_deleter<rows_entry>()::{lambda(rows_entry*)#1}) at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/intrusive_set_external_comparator.hh:210
 (inlined by) mutation_partition::clear_gently(cache_tracker*) at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/mutation_partition.cc:2303
remove_or_mark_as_unique_owner(partition_version*, mutation_cleaner*) at partition_version.cc:?
 (inlined by) mutation_cleaner_impl::destroy_gently(partition_version&) at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/mutation_cleaner.hh:84
 (inlined by) mutation_cleaner::destroy_gently(partition_version&) at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/mutation_cleaner.hh:156
 (inlined by) remove_or_mark_as_unique_owner at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/partition_version.cc:38
row_cache::invalidate_unwrapped(nonwrapping_range<dht::ring_position> const&) at crtstuff.c:?
 (inlined by) operator()<cache_entry*> at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/row_cache.cc:1133
 (inlined by) erase_and_dispose<row_cache::invalidate_unwrapped(const partition_range&)::<lambda()>::<lambda(auto:108&&)> > at /usr/include/boost/intrusive/bstree.hpp:1500
 (inlined by) private_erase<row_cache::invalidate_unwrapped(const partition_range&)::<lambda()>::<lambda(auto:108&&)> > at /usr/include/boost/intrusive/bstree.hpp:2111
 (inlined by) erase_and_dispose<row_cache::invalidate_unwrapped(const partition_range&)::<lambda()>::<lambda(auto:108&&)> > at /usr/include/boost/intrusive/bstree.hpp:1540
 (inlined by) operator() at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/row_cache.cc:1135
 (inlined by) with_allocator<row_cache::invalidate_unwrapped(const partition_range&)::<lambda()> > at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/./utils/allocation_strategy.hh:307
 (inlined by) row_cache::invalidate_unwrapped(nonwrapping_range<dht::ring_position> const&) at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/row_cache.cc:1130
std::_Function_handler<seastar::future<> (), row_cache::invalidate(seastar::noncopyable_function<void ()>, std::vector<nonwrapping_range<dht::ring_position>, std::allocator<nonwrapping_range<dht::ring_position> > >&&)::{lambda()#1}>::_M_invoke(std::_Any_data const&) at row_cache.cc:?
 (inlined by) with_linearized_managed_bytes<row_cache::invalidate(row_cache::external_updater, dht::partition_range_vector&&)::<lambda()>::<lambda()> > at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/./utils/managed_bytes.hh:428
 (inlined by) operator() at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/row_cache.cc:1110
 (inlined by) _M_invoke at /usr/include/c++/8/bits/std_function.h:283
_ZZN9row_cache9do_updateEN7seastar20noncopyable_functionIFvvEEESt8functionIFNS0_6futureIJEEEvEEENUlT_E0_clINS0_15semaphore_unitsINS0_35semaphore_default_exception_factoryENSt6chrono3_V212steady_clockEEEEEDaS9_ at row_cache.cc:?
 (inlined by) operator() at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/row_cache.cc:1272
 (inlined by) apply<row_cache::do_update(row_cache::external_updater, row_cache::internal_updater)::<lambda(auto:109)> mutable [with auto:109 = seastar::semaphore_units<>]::<lambda()> > at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/include/seastar/core/future.hh:1402
 (inlined by) futurize_apply<row_cache::do_update(row_cache::external_updater, row_cache::internal_updater)::<lambda(auto:109)> mutable [with auto:109 = seastar::semaphore_units<>]::<lambda()> > at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/include/seastar/core/future.hh:1474
 (inlined by) operator()<seastar::semaphore_units<> > at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/row_cache.cc:1279
row_cache::do_update(seastar::noncopyable_function<void ()>, std::function<seastar::future<> ()>) at crtstuff.c:?
 (inlined by) apply<row_cache::do_update(row_cache::external_updater, row_cache::internal_updater)::<lambda(auto:109)>, seastar::semaphore_units<seastar::semaphore_default_exception_factory, std::chrono::_V2::steady_clock> > at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/include/seastar/core/apply.hh:43
 (inlined by) apply<row_cache::do_update(row_cache::external_updater, row_cache::internal_updater)::<lambda(auto:109)>, seastar::semaphore_units<seastar::semaphore_default_exception_factory, std::chrono::_V2::steady_clock> > at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/include/seastar/core/future.hh:1392
 (inlined by) then_impl<row_cache::do_update(row_cache::external_updater, row_cache::internal_updater)::<lambda(auto:109)> > at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/include/seastar/core/future.hh:936
 (inlined by) then<row_cache::do_update(row_cache::external_updater, row_cache::internal_updater)::<lambda(auto:109)> > at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/include/seastar/core/future.hh:917
 (inlined by) row_cache::do_update(seastar::noncopyable_function<void ()>, std::function<seastar::future<> ()>) at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/row_cache.cc:1280
row_cache::invalidate(seastar::noncopyable_function<void ()>, std::vector<nonwrapping_range<dht::ring_position>, std::allocator<nonwrapping_range<dht::ring_position> > >&&) at crtstuff.c:?
row_cache::invalidate(seastar::noncopyable_function<void ()>, nonwrapping_range<dht::ring_position> const&) at crtstuff.c:?
table::add_sstable_and_update_cache(seastar::lw_shared_ptr<sstables::sstable>) at crtstuff.c:?
seastar::continuation<seastar::future<> seastar::future<>::then_impl<streaming::stream_session::init_messaging_service_handler()::{lambda(seastar::rpc::client_info const&, utils::UUID, utils::UUID, utils::UUID, unsigned long, seastar::rpc::optional<streaming::stream_reason>, seastar::rpc::source<frozen_mutation_fragment, seastar::rpc::optional<streaming::stream_mutation_fragments_cmd> >)#4}::operator()(seastar::rpc::client_info const&, utils::UUID, utils::UUID, utils::UUID, unsigned long, seastar::rpc::optional<streaming::stream_reason>, seastar::rpc::source<frozen_mutation_fragment, seastar::rpc::optional<streaming::stream_mutation_fragments_cmd> >) const::{lambda()#1}::operator()()::{lambda(seastar::lw_shared_ptr<schema const>)#1}::operator()(schema)::{lambda(flat_mutation_reader)#2}::operator()(seastar::lw_shared_ptr<schema const>) const::{lambda(bool)#1}::operator()(bool)::{lambda()#2}, seastar::future<> >(streaming::stream_session::init_messaging_service_handler()::{lambda(seastar::rpc::client_info const&, utils::UUID, utils::UUID, utils::UUID, unsigned long, seastar::rpc::optional<streaming::stream_reason>, seastar::rpc::source<frozen_mutation_fragment, seastar::rpc::optional<streaming::stream_mutation_fragments_cmd> >)#4}::operator()(seastar::rpc::client_info const&, utils::UUID, utils::UUID, utils::UUID, unsigned long, seastar::rpc::optional<streaming::stream_reason>, seastar::rpc::source<frozen_mutation_fragment, seastar::rpc::optional<streaming::stream_mutation_fragments_cmd> >) const::{lambda()#1}::operator()()::{lambda(seastar::lw_shared_ptr<schema const>)#1}::operator()(schema)::{lambda(flat_mutation_reader)#2}::operator()(seastar::lw_shared_ptr<schema const>) const::{lambda(bool)#1}::operator()(bool)::{lambda()#2}&&)::{lambda()#1}::operator()() const::{lambda({lambda(bool)#1})#1}>::run_and_dispose() at stream_session.cc:?
 (inlined by) apply at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/include/seastar/core/apply.hh:35
 (inlined by) apply<streaming::stream_session::init_messaging_service_handler()::<lambda(const seastar::rpc::client_info&, streaming::stream_session::UUID, streaming::stream_session::UUID, streaming::stream_session::UUID, uint64_t, seastar::rpc::optional<streaming::stream_reason>, seastar::rpc::source<frozen_mutation_fragment, seastar::rpc::optional<streaming::stream_mutation_fragments_cmd> >)>::<lambda()> mutable::<lambda(schema_ptr)> mutable::<lambda(flat_mutation_reader)>::<lambda(bool)> mutable::<lambda()> > at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/include/seastar/core/apply.hh:43
 (inlined by) apply<streaming::stream_session::init_messaging_service_handler()::<lambda(const seastar::rpc::client_info&, streaming::stream_session::UUID, streaming::stream_session::UUID, streaming::stream_session::UUID, uint64_t, seastar::rpc::optional<streaming::stream_reason>, seastar::rpc::source<frozen_mutation_fragment, seastar::rpc::optional<streaming::stream_mutation_fragments_cmd> >)>::<lambda()> mutable::<lambda(schema_ptr)> mutable::<lambda(flat_mutation_reader)>::<lambda(bool)> mutable::<lambda()> > at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/include/seastar/core/future.hh:1392
 (inlined by) operator()<seastar::future_state<> > at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/include/seastar/core/future.hh:950
 (inlined by) run_and_dispose at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/include/seastar/core/future.hh:377
seastar::reactor::run_tasks(seastar::reactor::task_queue&) at /usr/include/boost/program_options/variables_map.hpp:146
seastar::reactor::run_some_tasks() at /usr/include/boost/program_options/variables_map.hpp:146
seastar::reactor::run_some_tasks() at /usr/include/boost/program_options/variables_map.hpp:146
 (inlined by) seastar::reactor::run() at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/build/release/../../src/core/reactor.cc:4246
seastar::smp::configure(boost::program_options::variables_map, seastar::reactor_config)::{lambda()#3}::operator()() const at /usr/include/boost/program_options/variables_map.hpp:146
std::function<void ()>::operator()() const at /usr/include/c++/8/bits/std_function.h:687
 (inlined by) seastar::posix_thread::start_routine(void*) at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/build/release/../../src/core/posix.cc:52
?? ??:0
?? ??:0
```

**Comment by tgrabiec** [comment]
Duplicate of https://github.com/scylladb/scylla/issues/2683

Fixed in master.

**Comment by tgrabiec** [comment]
@bentsi Stalls which contain `invalidate_unwrapped()` are #2683

**Comment by slivne** [comment]
Fixed in master for 3.2
